CI: 单元测试和集成测试失败 (2026-03-06)

## 测试失败摘要

CI 监控检测到主分支更新后有测试失败。

### 触发提交

- 6bd556b: Merge pull request #715 from hs3180/feat/issue-692-group-service-create-group
- 746c546: fix(nodes): properly dispose SkillAgent after next-step execution (Issue #716) (#718)
- a0f6a6a: merge: resolve conflicts with main
- 4b5a453: feat(feishu): add createGroup() method to GroupService (Issue #692)

---

## 1. 单元测试失败

**测试文件:** \`src/utils/task-state-manager.test.ts\`
**测试用例:** \`TaskStateManager > listTaskHistory > should list completed tasks\`

**错误信息:**
\`\`\`
AssertionError: expected 'Task 1' to be 'Task 2' // Object.is equality

Expected: "Task 2"
Received: "Task 1"

 ❯ src/utils/task-state-manager.test.ts:243:33
    241|       const history = await listManager.listTaskHistory();
    242|       expect(history.length).toBe(2);
    243|       expect(history[0].prompt).toBe('Task 2'); // Most recent first
       |                                 ^
    244|       expect(history[1].prompt).toBe('Task 1');
\`\`\`

**根本原因分析:**
测试期望历史记录按"最近优先"排序（Task 2 在前），但实际返回的是 Task 1 在前。这表明 \`listTaskHistory\` 方法的排序逻辑可能与预期不符，或者任务完成时间的记录顺序有问题。

---

## 2. 集成测试失败

**测试套件:** Use Case 2 - Task Execution
**测试用例:** Test 4: Text analysis task

**错误信息:**
\`\`\`
[FAIL] Request failed with HTTP 000
\`\`\`

**根本原因分析:**
HTTP 000 表示 curl 无法建立连接，可能原因：
1. **请求超时** - AI 模型响应时间超过测试脚本的超时设置
2. **服务器资源竞争** - 多个测试同时运行导致服务器资源不足
3. **API 限流** - AI 提供商 (GLM) 可能对请求频率有限制

从服务器日志可以看到：
- 请求 \`test-use-case-2-text-27638\` 只记录了一条消息接收日志
- 没有后续的响应日志，表明请求可能在等待 AI 响应时超时

**相关日志:**
\`\`\`
[2026-03-06 00:26:34.206 +0800] DEBUG: [Pilot] SDK message received
    context: "Pilot"
    chatId: "test-use-case-2-text-27638"
    messageCount: 1
    type: "text"
\`\`\`

---

## 建议修复

### 单元测试
1. 检查 \`TaskStateManager.listTaskHistory()\` 的排序逻辑
2. 确认任务完成时间的记录是否正确

### 集成测试
1. 增加测试超时时间（当前 60s 可能不够）
2. 添加请求重试机制
3. 考虑在测试之间添加延迟以避免 API 限流

---

## 测试环境

- Node.js: v24.8.0
- 测试框架: Vitest
- 集成测试超时: 60s
- REST 端口: 3099

🤖 Generated by CI Monitor scheduled task

--- a/src/utils/task-state-manager.test.ts
+++ b/src/utils/task-state-manager.test.ts
@@ -4,11 +4,14 @@
  * Issue #468: 任务控制指令 - deep task 执行管理
  */
 
-import { describe, it, expect, beforeEach, afterEach } from 'vitest';
+import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
 import * as fs from 'fs/promises';
 import * as path from 'path';
 import * as os from 'os';
 import { TaskStateManager, resetTaskStateManager } from './task-state-manager.js';
+
+// Fixed timestamp for deterministic testing
+const MOCK_TIME = 1709667600000; // 2024-03-05T15:00:00.000Z
 
 describe('TaskStateManager', () => {
   let tempDir: string;
@@ -17,6 +20,10 @@ describe('TaskStateManager', () => {
   beforeEach(async () => {
     // Reset singleton for each test
     resetTaskStateManager();
+
+    // Use fake timers for deterministic time-based testing
+    vi.useFakeTimers();
+    vi.setSystemTime(MOCK_TIME);
 
     // Create a temporary directory for each test
     tempDir = await fs.mkdtemp(path.join(os.tmpdir(), 'task-state-test-'));
@@ -27,6 +34,9 @@ describe('TaskStateManager', () => {
     // Clean up the temporary directory
     await fs.rm(tempDir, { recursive: true, force: true });
     resetTaskStateManager();
+
+    // Restore real timers
+    vi.useRealTimers();
   });
 
   describe('startTask', () => {
@@ -232,15 +242,20 @@ describe('TaskStateManager', () => {
       const testDir = await fs.mkdtemp(path.join(os.tmpdir(), 'task-state-list-'));
       const listManager = new TaskStateManager(testDir);
 
+      // Task 1: Start and complete at MOCK_TIME
       await listManager.startTask('Task 1', 'oc_chat');
       await listManager.completeTask();
 
+      // Advance time by 1 second to ensure different timestamps
+      vi.advanceTimersByTime(1000);
+
+      // Task 2: Start and cancel at MOCK_TIME + 1000ms
       await listManager.startTask('Task 2', 'oc_chat');
       await listManager.cancelTask();
 
       const history = await listManager.listTaskHistory();
       expect(history.length).toBe(2);
-      expect(history[0].prompt).toBe('Task 2'); // Most recent first
+      expect(history[0].prompt).toBe('Task 2'); // Most recent first (completed at later time)
       expect(history[1].prompt).toBe('Task 1');
 
       await fs.rm(testDir, { recursive: true, force: true });
@@ -251,14 +266,19 @@ describe('TaskStateManager', () => {
       const testDir = await fs.mkdtemp(path.join(os.tmpdir(), 'task-state-limit-'));
       const limitManager = new TaskStateManager(testDir);
 
-      // Create 5 tasks
+      // Create 5 tasks with 1 second intervals for deterministic ordering
       for (let i = 0; i < 5; i++) {
         await limitManager.startTask(`Task ${i}`, 'oc_chat');
         await limitManager.completeTask();
+        vi.advanceTimersByTime(1000);
       }
 
       const history = await limitManager.listTaskHistory(3);
       expect(history.length).toBe(3);
+      // Most recent tasks first (Task 4, Task 3, Task 2)
+      expect(history[0].prompt).toBe('Task 4');
+      expect(history[1].prompt).toBe('Task 3');
+      expect(history[2].prompt).toBe('Task 2');
 
       await fs.rm(testDir, { recursive: true, force: true });
     });

